### ISSUE TITLE
[BUG] Codex hangs on start if Geth node not running

### ISSUE THREAD

**Issue by gmega** [codex-storage/nim-codex#726]
**Describe the bug**

[`waitForSync`](https://github.com/codex-storage/nim-codex/blob/6ee116298929c5419f055abda5eccb7e25caef11/codex/codex.nim#L90) got recently introduced and it probes the provided Geth node in a loop to make sure it is synced before Codex starts.

The problem is that if the Geth node is not running, the `isSyncing` predicate seems to freeze. Either we have a long timeout there (which will add up with the backoff mechanism in `waitForSync`), or there's a bug. Either way, this should die faster (i.e. shouldn't take minutes) if connections are being rejected.

**Comment by AuHau** [comment]
Duplicate of these:
- https://github.com/codex-storage/nim-codex/issues/705
- https://github.com/codex-storage/nim-codex/issues/640

But if you are working on it just keep this one and please link these issues in your PRs as well so they get closed.

**Comment by gmega** [comment]
Yep, already have a PR for it, thanks for letting me know about those.
